### ISSUE TITLE
qmail dash adressing

### ISSUE THREAD

**Issue by quite** [Uberspace/manual#528]
This is perhaps more of a support question. I'm writing here because it may interest others. And there could be a need for further documentation.

I'm using the older `.qmail-foo` file way of creating a number of email aliases, which if I understood correctly is deprecated?

But in particular I make much use of the possibility for arbitrary extension using a dash (`-`). This is similar to "plus addressing`. Over the years, I have created many `foo-something@fqdn` addresses, and used the, in various places (without keeping track of them). So my questions is, is there any way to move over to the newer way of creating email aliases, and still make use of this generic "dash addressing"?

**Comment by noave** [comment]
Yes this is rather a support question :-)  We will soon provide ways for migrating Mail configs like this. However, to show you a possible way right now, you could set up a catchall mailbox and then add a config with sieve like:

```
if envelope :regex "to" "(^foo\-)(.*)@(.*)" {
redirect "foo-default@example.com";
} else {
fileinto :create "catchall"; stop;
}
```

this redirects based on a regex (`foo-*@`) to a specific mailaddress and files everything else to a catchall box.

This would be the replacement for .qmail-foo-default files, for any other specific .qmail-foo-onething you should create mailusers.

**Comment by quite** [comment]
Thanks! So can we expect some further documentation on this at some point?

Also, do you have already a timeline for entirely removing support for .qmail-files?

**Comment by quite** [comment]
Thanks, good to know! I have a couple of comments regarding the documentation on these topics:

On the page https://manual.uberspace.de/mail-filters/#logging it says that I can look at file `~/users/$MAILBOX/.dovecot.sieve.log` if something does not work. But that file does not exist for the mail user which I'm now trying out sieve with. But I'm also not sure if the log file would list all mails processed by sieve, or only contain any errors (which I might not have caused).

At https://manual.uberspace.de/mail-mailboxes/#system-mailbox it says that I can remove the file `~/.qmail` if I want to have mails for `USERNAME@uber.space` dropped in `~/Maildir`, instead of redirected to my external email address. It would be useful if it also said what `~/.qmail` should contain to restore the latter functionality. (My `~/.qmail` currently contains `|maildrop`, but this ought to be a 3rd state, pertain to the legacy qmail-file filtering that I'm still using).

**Comment by quite** [comment]
I have managed to migrate the dash addressing to use Sieve, so for me all is good.

I'm still wondering about `~/users/$MAILBOX/.dovecot.sieve.log` which I don't have (any) -- it would be useful for debugging sieve!
